samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+                feedbackVertexSymbol = symbol
+                selectedVertexSymbol = symbol
+                vertexTextSymbol = null
+            }
+
+            // Hide the selected feature
+            (feature.featureTable?.layer as? FeatureLayer)?.setFeatureVisible(feature, false)
+
+            // Start the geometry editor and center the map underneath the reticle
+            feature.geometry?.let { initialGeometry ->
+                viewModelScope.launch {
+                    mapViewProxy.setViewpointCenter(initialGeometry.extent.center)
+                }
+                geometryEditor.start(initialGeometry)
+                geometryEditor.selectVertex(0,0)
+            }
+        }
+    }
+
+    /**
+     * Stop the geometry editor and discard the changes made to the geometry.
+     */
+    fun discardGeometryChanges() {
+        // Discard the current edit
+        geometryEditor.stop()
+
+        // Reset the selection
+        resetSelections()
+    }
+
+    /**
+     * Stop the geometry editor, and update the previously identified feature with the new geometry.
+     */
+    fun saveGeometryChanges() {
+        // Stop the geometry editor and get the updated geometry
+        val finalGeometry = geometryEditor.stop()
+
+        // Update the feature with the new geometry
+        selectedFeature?.let { feature ->
+            feature.geometry = finalGeometry
+
+            viewModelScope.launch {
+                (feature.featureTable as GeodatabaseFeatureTable).updateFeature(feature)
+                    .onFailure { error ->
+                        messageDialogVM.showMessageDialog(
+                            "Error updating feature",
+                            error.message.toString()
+                        )
+                    }
+            }
+        }
+
+        // Reset the selection
+        resetSelections()
+    }
+
+    /**
+     * Identifies the tapped screen coordinate in the provided [singleTapConfirmedEvent] and gets
+     * the asset at that location.
+     */
+    fun identify(singleTapConfirmedEvent: SingleTapConfirmedEvent) {
+
+        if (geometryEditor.isStarted.value || arcGISMap.operationalLayers.isEmpty()) {
+            return
+        }
+        viewModelScope.launch {
+            mapViewProxy.identifyLayers(
+                screenCoordinate = singleTapConfirmedEvent.screenCoordinate,
+                tolerance = 12.dp,
+                returnPopupsOnly = false,
+                maximumResults = 1
+            ).onSuccess { identifyResultList ->
+                // As we are using subtype feature layers in this sample the returned features are
+                // contained in the sublayer results
+                val identifiedFeature = identifyResultList.firstOrNull()?.sublayerResults?.firstOrNull()?.geoElements?.firstOrNull()
+                if (identifiedFeature !is ArcGISFeature) {
+                    resetSelections()
+                    return@launch
```

Comment:
Could be formatted as below and effectively is the same behavior: 
```suggestion
                    return@launch resetSelections()
```